sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ExtRefService.java

```diff
@@ -139,24 +136,23 @@ public static SclReport createDataSetAndControlBlocks(SCL scd, String targetIedN
      * @param scd               input SCD object. The object will be modified with the new DataSet and ControlBlocks
      * @param targetIedName     the name of the IED where the ExtRef are
      * @param targetLDeviceInst the name of the LDevice where the ExtRef are
-     * @return a report with all encountered errors
+     * @return list of encountered errors
      */
-    public static SclReport createDataSetAndControlBlocks(SCL scd, String targetIedName, String targetLDeviceInst) {
+    public static List<SclReportItem> createDataSetAndControlBlocks(SCL scd, String targetIedName, String targetLDeviceInst) {
         if (StringUtils.isBlank(targetIedName)) {
             throw new ScdException(MESSAGE_MISSING_IED_NAME_PARAMETER);
         }
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
         IEDAdapter iedAdapter = sclRootAdapter.getIEDAdapterByName(targetIedName);
         LDeviceAdapter lDeviceAdapter = iedAdapter.getLDeviceAdapterByLdInst(targetLDeviceInst);
-        return createDataSetAndControlBlocks(sclRootAdapter, Stream.of(lDeviceAdapter));
+        return createDataSetAndControlBlocks(Stream.of(lDeviceAdapter));
     }
 
-    private static SclReport createDataSetAndControlBlocks(SclRootAdapter sclRootAdapter, Stream<LDeviceAdapter> lDeviceAdapters) {
-        List<SclReportItem> sclReportItems = lDeviceAdapters
+    private static List<SclReportItem> createDataSetAndControlBlocks(Stream<LDeviceAdapter> lDeviceAdapters) {
+        return lDeviceAdapters
                 .map(LDeviceAdapter::createDataSetAndControlBlocks)
                 .flatMap(List::stream)
-                .toList();
-        return new SclReport(sclRootAdapter, sclReportItems);
+                .collect(Collectors.toList());
```

Comment:
Use `.toList()`. Caller should not assume that the returned List is modifiable.

Reply:
done